### DISCUSSION TITLE
Making the gutterView font size always follow the text size in its corresponding line.

### DISCUSSION THREAD

**Discussion by HassanTaleb90** [krzyzanowskim/STTextView#66] [Questions]
I think the current approach for the gutterView font size is not the best:

```
    /// The font used to draw line numbers.
    ///
    /// Initialized with a textView font value and does not update automatically when
    /// text view font changes.
    @Invalidating(.display)
    public var font: NSFont = adjustGutterFont(NSFont(descriptor: NSFont.monospacedDigitSystemFont(ofSize: 0, weight: .regular).fontDescriptor.withSymbolicTraits(.condensed), size: 0)!)
```
  

I think it would be better to make the gutterView font size always follow the text size in its corresponding line to prevent bugs like the following:

Try changing the font size in ViewController to a large one, let's say 288:
`textView.font = NSFont.monospacedSystemFont(ofSize: 288, weight: .regular)`
You will notice the bug shown in the images.

<img width="570" alt="Screenshot 2024-08-23 at 7 52 46 PM" src="https://github.com/user-attachments/assets/5db72bea-ff00-4986-a4a7-e3f0de620556">
<img width="610" alt="Screenshot 2024-08-23 at 7 53 00 PM" src="https://github.com/user-attachments/assets/0dc3c162-99aa-48c2-ae79-1fefd4fd2ef9">

https://github.com/user-attachments/assets/9ab7a210-a2be-4eda-b683-980e06565ae7

**Comment by krzyzanowskim** [comment]
the gutter font is initialized with the textview font (default font), including it's size.
I agree. It may not be the best thing. Maybe it should not automatically use the textview font and leave that to the user that can still adjust the gutter font manually.

**Comment by krzyzanowskim** [comment]
changed in f61e63a4e04d7e9f057b5ced50a7a70596620c55

**Comment by HassanTaleb90** [comment]
I think it would be better if each line number automatically matches the font size of the largest character on the corresponding line.

8 points font size:
<img width="229" alt="Screenshot 2024-08-23 at 8 43 06 PM" src="https://github.com/user-attachments/assets/aaf2fa93-ca1d-45ff-8950-0184eb65b8c7">

https://github.com/user-attachments/assets/8b5fcdd3-7911-488e-a8e7-c78dfece8e5e

Xcode:

https://github.com/user-attachments/assets/b11ca501-ef71-4805-906c-68b6370e86ae

**Reply by krzyzanowskim** [reply]
> if each line number automatically matches the font size of the largest character on the corresponding line.

I don't like it. It makes the line numbers of different sizes, nothing is aligned, each line is different, and there is too much noise. 

What is on the Xcode recording here is that it scales the line numbers and the content font, not matching them to what's in the line - slightly different behavior that actually makes sense.

**Reply by krzyzanowskim** [reply]
> What is on the Xcode recording here is that it scales the line numbers and the content font

that is Xcode-specific behavior. Normally, only selected text is scaled (compared with TextEdit). That means only selected lines are affected. The scaling like in Xcode should be handled outsite STTextView. That said, I don't feel to change anything in that regard now.
